src/Api/Management/Entity/App.php

```diff
@@ -216,4 +219,33 @@ public function setScopes(string ...$scopes): void
     {
         $this->privateSetScopes(...$scopes);
     }
+
+    /**
+     * Get Initial API Products only during app creation.
+     *
+     * @return array $initialApiProducts
+     *
+     * @internal
+     */
+    final public function getApiProducts(): array
+    {
+        return $this->initialApiProducts;
+    }
+
+    /**
+     * Set Initial API Products only during app creation.
+     * The method is not supported to update existing App.
+     *
+     * @param array $initialApiProducts
+     *
+     * @throws \Exception If used to update existing App.
+     */
+    final public function setInitialApiProducts(array $initialApiProducts): void
+    {
+        if (!$this->appId) {
+            $this->initialApiProducts = $initialApiProducts;
+        } else {
+            throw new \Exception('This method is only supported for creating a new app.');
```

Comment:
Should be a \LogicException() since calling it is a "mistake made by the developer". 

(Although we know this method should only exist on New app entities for which we do not have an object type...)

Reply:
This has been taken care. Thank you for pointing it out.